module/VuFind/src/VuFind/View/Helper/Root/Metadata.php

```diff
@@ -0,0 +1,107 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Metadata_Vocabularies
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+/**
+ * Metadata view helper
+ *
+ * @category VuFind
+ * @package  Metadata_Vocabularies
+ * @author   Mario Trojan <mario.trojan@uni-tuebingen.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Metadata extends \Zend\View\Helper\AbstractHelper
+{
+    /**
+     * Metadata configuration entries
+     *
+     * @var \Zend\Config\Config
+     */
+    protected $config;
+
+    /**
+     * Zend meta helper, used to embed html tags in the generated page
+     *
+     * @var \Zend\View\Helper\HeadMeta
+     */
+    protected $metaHelper;
+
+    /**
+     * Plugin Manager for vocabularies
+     *
+     * @var \VuFind\MetadataVocabulary\PluginManager
+     */
+    protected $pluginManager;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFind\MetadataVocabulary\PluginManager $pm         Plugin manager
+     * @param \Zend\Config\Config                      $config     Configuration
+     * @param \Zend\View\Helper\HeadMeta               $metaHelper Head meta helper
+     */
+    public function __construct(\VuFind\MetadataVocabulary\PluginManager $pm,
+        \Zend\Config\Config $config,
+        \Zend\View\Helper\HeadMeta $metaHelper
+    ) {
+        $this->pluginManager = $pm;
+        $this->config = $config;
+        $this->metaHelper = $metaHelper;
+    }
+
+    /**
+     * Generate all metatags for RecordDriver and add to page
+     *
+     * Decide which Plugins to load for the given RecordDriver
+     * dependant on configuration. (only by class name,
+     * namespace will not be considered)
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $driver Record driver
+     *
+     * @return void
+     */
+    public function generateMetatags(\VuFind\RecordDriver\AbstractBase $driver)
+    {
+        $recordDriverConfigs = $this->config->Vocabularies ?? [];
+        foreach ($recordDriverConfigs as $recordDriverClassName => $metatagTypes) {
+            if ($driver instanceof $recordDriverClassName) {
+                foreach ($metatagTypes as $metatagType) {
+                    $vocabulary = $this->pluginManager->get($metatagType);
+                    $mappedFields = $vocabulary->getMappedData($driver);
+                    foreach ($mappedFields as $field => $values) {
+                        foreach ($values as $value) {
+                            $this->metaHelper->appendName($field, $value);
+                        }
+                    }
+                }
+                break;
```

Comment:
I ran into a problem where if I defined multiple vocabularies using different class names (i.e. one vocabulary for SolrMarc, and a different one for AbstractBase), only the first one to match would be used... but I think the expected behavior would be to get all of them since they all apply. I ended up refactoring this code a little bit to resolve the problem; I've opened another pull request: https://github.com/ubtue/tuefind/pull/1058